test/k8sT/Policies.go

```diff
@@ -88,197 +80,421 @@ var _ = Describe("K8sPolicyTest", func() {
+			kubectl.Delete(kubectl.ManifestGet(webPolicy))
+			kubectl.Delete(kubectl.ManifestGet(redisPolicy))
+			kubectl.Delete(kubectl.ManifestGet(redisPolicyDeprecated))
+		})
+
+		waitforPods := func() {
+			pods, err := kubectl.WaitforPods(
+				helpers.DefaultNamespace,
+				fmt.Sprintf("-l %s", groupLabel), 300)
+			ExpectWithOffset(1, pods).Should(BeTrue())
+			ExpectWithOffset(1, err).Should(BeNil())
+		}
+
+		testConnectivitytoRedis := func() {
+			webPods, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, "-l k8s-app.guestbook=web")
+			Expect(err).To(BeNil(), "Cannot get web pods")
+
+			for pod := range webPods {
+				command := `nc redis-master 6379 <<EOF
+							PING
+							EOF`
+				_, err := kubectl.ExecPodCmd(helpers.DefaultNamespace, pod, command)
+				ExpectWithOffset(1, err).To(BeNil(), "Web pod %q cannot connect to redis", pod)
+			}
+		}
+		It("checks policy example", func() {
+			waitforPods()
+
+			By("Apply policy to web")
+			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+			_, err := kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, kubectl.ManifestGet(webPolicy),
+				helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "Cannot apply web-policy err: %q", err)
+
+			err = waitUntilEndpointUpdates(ciliumPod, eps, 3)
+			Expect(err).To(BeNil(), "Pods are not ready after timeout")
+
+			policyCheck := fmt.Sprintf("cilium policy get %s=%s %s=%s",
+				helpers.KubectlPolicyNameLabel, webPolicyName,
+				helpers.KubectlPolicyNameSpaceLabel, helpers.DefaultNamespace)
+			kubectl.CiliumExec(ciliumPod, policyCheck).ExpectSuccess(
+				"Policy %q is not in cilium", webPolicyName)
+			kubectl.CiliumExec(ciliumPod2, policyCheck).ExpectSuccess(
+				"Policy %q is not in cilium", webPolicyName)
+
+			By("Apply policy to Redis")
+			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, kubectl.ManifestGet(redisPolicy),
+				helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "Cannot apply redis policy err:%q", err)
+
+			err = waitUntilEndpointUpdates(ciliumPod, eps, 1)
+			Expect(err).To(BeNil(), "Pods are not ready after timeout")
+
+			policyCheck = fmt.Sprintf("cilium policy get %s=%s %s=%s",
+				helpers.KubectlPolicyNameLabel, redisPolicyName,
+				helpers.KubectlPolicyNameSpaceLabel, helpers.DefaultNamespace)
+
+			kubectl.CiliumExec(ciliumPod, policyCheck).ExpectSuccess(
+				"Policy %q is not in cilium", redisPolicyName)
+			kubectl.CiliumExec(ciliumPod2, policyCheck).ExpectSuccess(
+				"Policy %q is not in cilium", redisPolicyName)
+
+			testConnectivitytoRedis()
+
+			kubectl.Delete(kubectl.ManifestGet(redisPolicy)).ExpectSuccess(
+				"Cannot delete the redis policy")
+
+			By("Apply deprecated policy to Redis")
+
+			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, kubectl.ManifestGet(redisPolicyDeprecated),
```

Comment:
The bash test only imports the deprecated Redis policy if we are running Kubernetes 1.6. Won't we be importing the deprecated policy for each K8s version we use?

Reply:
I think that the both version should be tested.  Both should work in kubernetes 1.7 and we should be sure that it's tested correctly ;-)

Reply:
You're right, I actually misread the bash test when looking side-by-side.